driver/pace/driver/grid.py

```diff
@@ -0,0 +1,273 @@
+        communicator: CubedSphereCommunicator,
+    ) -> Tuple[DampingCoefficients, DriverGridData, GridData]:
+        return self.config.get_grid(
+            quantity_factory=quantity_factory, communicator=communicator
+        )
+
+    @classmethod
+    def from_dict(cls, config: dict):
+        instance = cls.registry.from_dict(config)
+        return cls(config=instance, type=config["type"])
+
+
+@GridInitializerSelector.register("generated")
+@dataclasses.dataclass
+class GeneratedConfig(GridInitializer):
+    """
+    Configuration for baroclinic initialization.
+
+    Attributes:
+        stretch_grid: whether to Schmidt transform the grid
+            (local refinement)
+        stretch_factor: refinement amount
+        lon_target: desired center longitude for refined tile (deg)
+        lat_target: desired center latitude for refined tile (deg)
+        tc_ks: something to do with friction in the vertical ???Ajda
+        restart_path: path to restart data
+    """
+
+    stretch_grid: bool = False
+    stretch_factor: Optional[float] = None
+    lon_target: Optional[float] = None
+    lat_target: Optional[float] = None
+    ks: int = 0
+    restart_path: Optional[
+        str
+    ] = None  # can this be just path from config? can I see that?
+    # if restart_path, then read in vertical grid
+
+    def __post_init__(self):
+        if self.stretch_grid:
+            if not self.stretch_factor:
+                raise ValueError(
+                    "Stretch_mode is true, but no stretch_factor is provided."
+                )
+            if not self.lon_target:
+                raise ValueError("Stretch_grid is true, but no lon_target is provided.")
+            if not self.lat_target:
+                raise ValueError("Stretch_grid is true, but no lat_target is provided.")
+
+    def get_grid(
+        self,
+        quantity_factory: QuantityFactory,
+        communicator: CubedSphereCommunicator,
+    ) -> Tuple[DampingCoefficients, DriverGridData, GridData]:
+
+        metric_terms = MetricTerms(
+            quantity_factory=quantity_factory, communicator=communicator
+        )
+        np = metric_terms.lat.np
+
+        if self.stretch_grid:  # do horizontal grid transformation
+            metric_terms = _transform_horizontal_grid(self, metric_terms)
+        grid_data = GridData.new_from_metric_terms(metric_terms)
+
+        if self.restart_path is not None:  # read in vertical grid
+            grid_data = _replace_vertical_grid(self, metric_terms)
+
+        damping_coefficients = DampingCoefficients.new_from_metric_terms(metric_terms)
+        driver_grid_data = DriverGridData.new_from_metric_terms(metric_terms)
+
+        return damping_coefficients, driver_grid_data, grid_data
+
+
+@GridInitializerSelector.register("serialbox")
+@dataclasses.dataclass
+class SerialboxConfig(GridInitializer):
+    """
+    Configuration for Serialbox initialization.
+    """
+
+    path: str
+    serialized_grid: bool
```

Comment:
You can remove the `serialized_grid` option, if we're using the SerialboxConfig to initialize the grid then we always want the serialized grid.